src/foam/mlang/mlang.js

```diff
@@ -1268,13 +1268,16 @@ foam.CLASS({
       name: 'arg2'
     },
     {
+      class: 'Map',
       name: 'groups',
-      factory: function() { return {}; }
+      factory: function() { return {}; },
+      javaFactory: `return new java.util.HashMap<Object, foam.dao.Sink>();`
     },
     {
-      class: 'StringArray',
+      class: 'List',
       name: 'groupKeys',
-      factory: function() { return []; }
+      factory: function() { return []; },
+      javaFactory: `return new java.util.ArrayList();`
```

Comment:
The javaFactory should just be set in the List Model definition. I just did that and will remove this line.